src/dhcp/aca_dhcp_state_handler.cpp

```diff
@@ -174,9 +180,23 @@ int Aca_Dhcp_State_Handler::update_dhcp_state_workitem_v2(const DHCPState curren
                   subnet_id.c_str());
     overall_rc = EXIT_FAILURE;
   }
-
+  alcor::schema::OperationType current_operation_type =
+          current_DhcpState.operation_type();
+  if (current_operation_type == alcor::schema::OperationType::UPDATE) {
+    for (auto &port_state : parsed_struct.port_states()) {
```

Comment:
for the goalstatev2, you may direct access the corresponding port_state assuming the port resource ID is the same as the DHCP resource ID. See this for an example: https://github.com/futurewei-cloud/alcor-control-agent/blob/817a65218a27e53475e66addb7b1e6fc9c3010a0/src/dp_abstraction/aca_dataplane_ovs.cpp#L380

Reply:
This is really a good way to do it, thanks for pointing it out 👍